Installer.hs

```diff
@@ -72,7 +70,25 @@ installer arch quick version = nsis $ do
         -- make sure we don't remove $APPDATA/cabal/bin, since users may have had that on their $PATH before
         mapM_ (setEnvVarRemove HKCU "PATH") $ "$INSTDIR/switch" : tail path
 
+    where
+        switcherNameSuffixes
+            = "" -- no suffix
+            : ['-' : intercalate "-" parts
+              | parts <- [[showArchAbbr arch]
+                         ,[version GHC]
+                         ,[version GHC, showArchAbbr arch]
+                         ,[majorVersion (version GHC)]
+                         ,[majorVersion (version GHC), showArchAbbr arch]]
+              ]
 
 showArchAbbr :: Arch -> String
 showArchAbbr Arch32 = "32"
 showArchAbbr Arch64 = "64"
+
+
+majorVersion :: Version -> Version
+majorVersion ver =
+    if length parts >= 3
```

Comment:
You're right. I didn't realize that. `head` has always left me in a state of "partial paranoia" (pun intended).